site/content/features/stringtemplates.md

```diff
@@ -0,0 +1,466 @@
+
+      // 2. Text block template
+      String div = STR."""
+<div>
+   <p class="greeting">Hello, \{name}!</p>
+   <p>Next year, you'll be \{age + 1}.</p>
+</div>""";
+      System.out.println(div);
+
+      // 3. String literal
+      message = STR."Hello, TODO! Next year, you'll be TODO.";
+      System.out.println(message);
+
+      // 4. Text block
+      div = STR."""
+<div>
+   <p class="greeting">Hello, TODO!</p>
+   <p>Next year, you'll be TODO.</p>
+</div>""";
+      System.out.println(div);
+   }
+}
+
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+
+# Predefined Template Processors
+
+The Java 21 API contains three template processors: the interpolating `STR` processor that you have already seen, as well as processors for formatting and for obtaining an unprocessed template.
+
+The `STR` processor, a static field of the class `java.lang.StringTemplate` is special—it is always imported. In other words, every Java file has these automatic imports:
+
+```
+import java.lang.*;
+import static java.lang.StringTemplate.STR;
+```
+
+The other processors must be imported manually:
+
+```
+import static java.util.FormatProcessor.FMT;
+import static java.lang.StringTemplate.RAW;
+```
+
+The `FMT` processor carries out `printf` style formatting:
+
+```
+String line = FMT."%-20s\{item.description} | %5d\{item.quantity} | %10.2f\{item.price}%n";
+```
+
+The format specifiers must immediately precede the embedded expressions.
+
+The `RAW` processor yields the template argument as a `StringTemplate` instance:
+
+```
+StringTemplate template = RAW."Hello, \{name}! Next year, you'll be \{age + 1}.";
+```
+
+See the following section for more information about the `StringTemplate` class.
+
+Note that a template processor can produce an object of any class. The `FMT` processor yields a `String`, but the `RAW` processor yields an object of the class `StringTemplate`.
+
+Here is a sandbox with these examples. Try adding a space after a format specifier. Also try incrementing `age` after the template with the embedded expression `\{age + 1}` was formed. Is the value updated? Should it be?
+
+{{< sandbox version=java19 preview="true" mainclass="TemplateProcessors" >}}{{< sandboxsource "TemplateProcessors.java" >}}
+import static java.util.FormatProcessor.FMT;
+import static java.lang.StringTemplate.RAW;
+
+public class TemplateProcessors {
+   record Item(String description, int quantity, double price) {}
+
+   public static void main(String[] args) {
+      String name = "Fred";
+      int age = 42;
+      var item = new Item("Blackwell Toaster", 2, 29.95);
+      String line = FMT."%-20s\{item.description} | %5d\{item.quantity} | %10.2f\{item.price}%n";
+      System.out.print(line);
+      item = new Item("Zappa Microwave Oven", 1, 109.95);
+      line = FMT."%-20s\{item.description} | %5d\{item.quantity} | %10.2f\{item.price}%n";
+      System.out.print(line);
+      StringTemplate template = RAW."Hello, \{name}! Next year, you'll be \{age + 1}.";
```

Comment:
Good point. Maybe this can also be shown with the `Item` instance, e.g.

    StringTemplate template = RAW."VAT for \{item.description} is \{item.price * 0.20}.";

Reply:
That's not quite the same thing. The question is when the expressions get evaluated (i.e. when obtaining the template or when processing it). I changed the instructions to assigning another object to `item`.